### PR TITLE
implemented show detailed information feature

### PR DESCRIPTION
**Pull request by Faka-GLB**
Created a DialogFragment to show detailed information about the weather item clicked.

Nexus 5x
image

Nexus s
image

Project tree
PronosticAR project tree

### CODE REVIEW

File: app/src/main/java/com/onboarding/pronosticAR/mvp/view/DetailedWeatherView.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            }
+        }
+    }
+
+    override fun returnToActivity() {
+        (fragment as DialogFragment).dismiss()
+    }
+}
+
+
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/pronosticAR/mvp/view/DetailedWeatherView.kt]: remove these unnecessary spaces pls

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/onboarding/pronosticAR/mvp/view/DetailedWeatherView.kt (hunk 1 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/pronosticAR/mvp/view/DetailedWeatherView.kt]: somehow ctrl+alt+l didin't deleted those lines

File: app/src/main/java/com/onboarding/pronosticAR/activity/MainActivity.kt [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        presenter = MainPresenter(model = MainModel(WeatherService()), view = MainView(this, binding))
+        presenter = MainPresenter(model = MainModel(WeatherService()), view = MainView(this, binding, this))
+    }
+
+    override fun onItemClick(weather: WeatherList) {
```
Comment (lucianoeli) [inline app/src/main/java/com/onboarding/pronosticAR/activity/MainActivity.kt]: I think this entity should be called  WeatherEntity right? its not a List... or something like ...ListItem

Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/pronosticAR/activity/MainActivity.kt]: renamed

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/res/layout/detailed_weather_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/text_view_detailed_weather_fragment_title"
+        style="@style/TextStyleDetailedWeatherFragmentTitle"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
```
Comment (mauroLonghi) [inline app/src/main/res/layout/detailed_weather_fragment.xml]: if you have mat_parent , not need constrainst guidelines. Removed please

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.onboarding.pronosticAR.contract
+
+import com.onboarding.pronosticAR.domain.entity.WeatherListItem
+
+interface DetailedWeatherContract {
+    interface DetailedWeatherPresenter {
+        fun onReturnButtonPressed()
+        fun initInformation(weatherItem: WeatherListItem?)
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt]: please avoid nullable lists

File: app/src/main/java/com/onboarding/pronosticAR/mvp/view/DetailedWeatherView.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                binding.textViewDetailedWeatherFragmentCloudness.text =
+                    it.getString(R.string.detailed_weather_fragment_text_view_cloudness, weather.clouds.all)
+                binding.textViewDetailedWeatherFragmentVisibility.text =
+                    it.getString(R.string.detailed_weather_fragment_text_view_visibility, weather.visibility)
+            }
+        }
+    }
+
+    override fun dismissFragment() {
+        (fragment as DialogFragment).dismiss()
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/pronosticAR/mvp/view/DetailedWeatherView.kt:38]: fragment is already DialogFragment

File: app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    }
+
+    interface DetailedWeatherView {
+        fun showDetailedWeatherInfo(weatherItem: WeatherListItem?)
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt]: please avoid nullable lists

File: app/src/main/res/layout/detailed_weather_fragment.xml (hunk 3 above)
Comment (Faka-GLB) [reply app/src/main/res/layout/detailed_weather_fragment.xml]: done

File: app/src/main/java/com/onboarding/pronosticAR/mvp/view/MainView.kt [hunk 7]
```diff
@@ -1,17 +1,31 @@
...
-        binding.recyclerViewMainActivity.adapter = WeatherAdapter(forecast.list)
+        binding.recyclerViewMainActivity.adapter = WeatherAdapter(forecast.listItem, weatherItemClickListener)
+    }
+
+    override fun showDetailedWeatherInfo(bundle: Bundle) {
+        val dialog = DetailedWeatherDialogFragment.newInstance(bundle)
+        activity?.let {
+            dialog.show((it as AppCompatActivity).supportFragmentManager, MAIN_VIEW_TAG)
+        }
+    }
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/pronosticAR/mvp/view/MainView.kt:27]: missing space below

File: app/src/main/res/layout/detailed_weather_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/text_view_detailed_weather_fragment_title"
+        style="@style/TextStyleDetailedWeatherFragmentTitle"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
```
Comment (nicolbidone) [inline app/src/main/res/layout/detailed_weather_fragment.xml:13]: replace this with  android:layout_width="match_parent"

File: app/src/main/java/com/onboarding/pronosticAR/mvp/view/MainView.kt (hunk 7 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/pronosticAR/mvp/view/MainView.kt:27]: added

File: app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt (hunk 6 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt]: done

File: app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt (hunk 4 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/pronosticAR/contract/DetailedWeatherContract.kt]: done

File: app/src/main/res/layout/detailed_weather_fragment.xml (hunk 8 above)
Comment (Faka-GLB) [reply app/src/main/res/layout/detailed_weather_fragment.xml:13]: with turkish we've figured out that solution becouse match_parent wasn't centering the TextView somehow

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**